server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -761,4 +763,12 @@ public IndexRequest setDynamicTemplates(Map<String, String> dynamicTemplates) {
     public Map<String, String> getDynamicTemplates() {
         return dynamicTemplates;
     }
+
+    public Object getRawTimestamp() {
+        return rawTimestamp;
+    }
+
+    public void setRawTimestamp(Object rawTimestamp) {
+        this.rawTimestamp = rawTimestamp;
```

Comment:
Maybe add `assert this.rawTimestamp == null` before setting `this.rawTimestamp`?